yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -356,8 +361,8 @@ private[spark] class Client(
     logInfo("Preparing resources for our AM container")
     // Upload Spark and the application JAR to the remote file system if necessary,
     // and add them as local resources to the application master.
-    val fs = FileSystem.get(hadoopConf)
-    val dst = getAppStagingDirPath(sparkConf, fs, appStagingDir)
+    val dst = new Path(appStagingBaseDir, appStagingDir)
```

Comment:
You could pass `appStagingDir` as a `Path` to this method and save some duplication; same for `setupLaunchEnv` below.

This whole class could use some cleanup in that regard, but these two are pretty low-hanging fruit.